From c6db04cec4e43a12d935b541bb389e8e250bf6f2 Mon Sep 17 00:00:00 2001
From: Lily Zhang <r58066@freescale.com>
Date: Tue, 26 Jul 2011 18:35:32 +0800
Subject: [PATCH] ENGR00153703 mc34708: remove generation id check

The generation ID of MC34708 Rev 2.3 is 0x13.
The generation ID of MC34708 Rev 2.1 is 0x91.
This patch is used to remove generation id check to support
different Rev chips.

Signed-off-by: Lily Zhang <r58066@freescale.com>
---
 drivers/mxc/pmic/core/pmic_core_i2c.c |   12 ------------
 1 files changed, 0 insertions(+), 12 deletions(-)

diff --git a/drivers/mxc/pmic/core/pmic_core_i2c.c b/drivers/mxc/pmic/core/pmic_core_i2c.c
index de5ca76..c839cb9 100644
--- a/drivers/mxc/pmic/core/pmic_core_i2c.c
+++ b/drivers/mxc/pmic/core/pmic_core_i2c.c
@@ -54,9 +54,6 @@
 #define MC13892_GEN_ID_VALUE	    0x7
 #define MC13892_IC_ID_VALUE		    1
 
-#define MC34708_GEN_ID_VALUE	    0x91
-#define MC34708_GENERATION_ID_LSH	0
-#define MC34708_GENERATION_ID_WID	9
 /*
  * Global variables
  */
@@ -200,15 +197,6 @@ static int __devinit is_chip_onboard(struct i2c_client *client)
 	/*bind the right device to the driver */
 	if (pmic_i2c_24bit_read(client, REG_IDENTIFICATION, &ret) == -1)
 		return -1;
-	if ((MC13892_GEN_ID_VALUE != BITFEXT(ret, MC13892_GENERATION_ID)) &&
-	   (MC34708_GEN_ID_VALUE != BITFEXT(ret, MC34708_GENERATION_ID))) {
-		/*compare the address value */
-		dev_err(&client->dev,
-			"read generation ID 0x%x is not equal to 0x%x!\n",
-			BITFEXT(ret, MC13892_GENERATION_ID),
-			MC13892_GEN_ID_VALUE);
-		return -1;
-	}
 
 	return 0;
 }
-- 
1.5.4.4

